### PR TITLE
Introduce gluster::logging

### PR DESCRIPTION
**Pull request by paladox**
This is to allow us to read files such as glusterd.log and send to gluster.

Bug: T5044

### CODE REVIEW

File: modules/gluster/manifests/logging.pp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#
+define gluster::logging (
+	Array[String] $file_source_options
+) {
+	syslog_ng::rewrite { 'r_application_name':
+		params => {
+			'type'      => 'set',
+			'options'   => [
+				'glusterd',
+				{ 'value' => 'APPLICATION_NAME' }
```
Comment (Southparkfan) [inline modules/gluster/manifests/logging.pp]: 'APPLICATION_NAME' must be 'PROGRAM'

File: modules/gluster/manifests/logging.pp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		},
+	} ->
+	syslog_ng::log { "s_file_${title} to d_graylog_syslog_tls":
+		params => [
+			{
+				'source' => "s_file_${title}",
+			},
+                        {
+                            'rewrite' => 'r_application_name',
+                        },
```
Comment (Southparkfan) [inline modules/gluster/manifests/logging.pp]: Indent

File: modules/gluster/manifests/logging.pp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			'options'   => [
+				'glusterd',
+				{ 'value' => 'APPLICATION_NAME' }
+			],
+		},
+	} ->
+	syslog_ng::source { "s_file_${title}":
+		params => {
+			'type'    => 'file',
+			'options' => $file_source_options,
```
Comment (Southparkfan) [inline modules/gluster/manifests/logging.pp:26]: I have given a VALID array of options below:

'options' => [
    '/var/log/glusterfs/glusterd.log',
    { 'flags' => 'no-parse' }
]

**Review by Southparkfan — DISMISSED** [review]

File: modules/gluster/manifests/logging.pp (hunk 3 above)
Comment (paladox) [reply modules/gluster/manifests/logging.pp:31]: Fixed!

File: modules/gluster/manifests/logging.pp (hunk 1 above)
Comment (paladox) [reply modules/gluster/manifests/logging.pp]: Fixed, thanks!

File: modules/gluster/manifests/logging.pp (hunk 2 above)
Comment (paladox) [reply modules/gluster/manifests/logging.pp]: Fixed, thanks!

**Pull request merged**